pkg/deployment/config.go

```diff
@@ -88,6 +88,19 @@ func NewConfigFromMap(configMap map[string]string) (*Config, error) {
 	nc := defaultConfig()
 
 	if err := cm.Parse(configMap,
+		// Legacy keys for backwards compatibility
+		cm.AsString("queueSidecarImage", &nc.QueueSidecarImage),
+		cm.AsDuration("progressDeadline", &nc.ProgressDeadline),
+		cm.AsDuration("digestResolutionTimeout", &nc.DigestResolutionTimeout),
+		cm.AsStringSet("registriesSkippingTagResolving", &nc.RegistriesSkippingTagResolving),
+		cm.AsQuantity("queueSidecarCPURequest", &nc.QueueSidecarCPURequest),
+		cm.AsQuantity("queueSidecarMemoryRequest", &nc.QueueSidecarMemoryRequest),
+		cm.AsQuantity("queueSidecarEphemeralStorageRequest", &nc.QueueSidecarEphemeralStorageRequest),
+		cm.AsQuantity("queueSidecarCPULimit", &nc.QueueSidecarCPULimit),
+		cm.AsQuantity("queueSidecarMemoryLimit", &nc.QueueSidecarMemoryLimit),
+		cm.AsQuantity("queueSidecarEphemeralStorageLimit", &nc.QueueSidecarEphemeralStorageLimit),
+		cm.AsString("concurrencyStateEndpoint", &nc.ConcurrencyStateEndpoint),
```

Comment:
Should we try to automate this by always parsing both keys in our parsing functions? We could potentially auto-transform them from either camelCase to kebab-case or back to spare us the double definition.

Reply:
nah - I think that would make the configmap.Parse API confusing